packages/react/src/hooks/accounts/useConnect.ts

```diff
@@ -35,35 +42,38 @@ export const useConnect = () => {
         }))
         const data = await connector.connect()
 
-        // Update connector globally only after successful connection
-        setGlobalState((x) => ({ ...x, connector, data }))
-        setLastUsedConnector(connector.name)
-        setState((x) => ({ ...x, loading: false }))
+        if (!didCancel) {
+          // Update connector globally only after successful connection
+          setGlobalState((x) => ({ ...x, connector, data }))
+          setLastUsedConnector(connector.name)
+          setState((x) => ({ ...x, loading: false }))
+        }
         return { data, error: undefined }
       } catch (error_) {
         const error = <Error>error_
-        setState((x) => ({ ...x, connector: undefined, error, loading: false }))
+        if (!didCancel) {
+          setState((x) => ({
+            ...x,
+            connector: undefined,
+            error,
+            loading: false,
+          }))
+        }
         return { data: undefined, error }
       }
     },
-    [globalState.connector, setGlobalState, setLastUsedConnector],
+    [cancelQuery, globalState.connector, setGlobalState, setLastUsedConnector],
   )
 
   // Keep connector in sync with global connector
   React.useEffect(() => {
-    let didCancel = false
-    if (didCancel) return
-
     setState((x) => ({
       ...x,
       connector: globalState.connector,
       error: undefined,
     }))
-
-    return () => {
-      didCancel = true
-    }
```

Comment:
And this can be safely removed since the only use of `didCancel` here is in the `if` check right after its definition.